### DISCUSSION TITLE
Calling a closure

### DISCUSSION THREAD

**Discussion by m-lima** [nushell/nushell#7503] [Q&A] [answered]
I'm wondering how can I call a closure that has been saved in a variable.

For example:
```
let hello = {|name| print $'Hello ($name)'}

# Call $hello somehow
$hello 'world'
```

Is there a way to do this at the moment?

**Comment by fdncred** [comment, marked as answer]
```
do $hello world
```

**Reply by theAkito** [reply]
Wow, this is fancy. Is this documented somewhere?

**Reply by kgadek** [reply]
Had same question, found this thread. While GitHub issue search is not the most ergonomic documentation type, it's great that good answers exist. Thanks all!

**Reply by J-Cake** [reply]
I must confess I find the Nushell docs rather unwieldly. It'd be great to have a reference somewhere with all the commands, all operators, syntax and behaviours clearly structured and documented. 

The existing documentation should be geared more towards a getting-started-style guide, much like how [node-gui](https://docs.nodegui.org/) does it. While it's not the best example because the docs aren't super thorough, I really appreciate the way its structured.

**Reply by fdncred** [reply]
We accept PRs. Nushell moves fast and keeping the docs updated is a never-ending process.

**Reply by J-Cake** [reply]
Ye I'm not experienced with nushell enough to be able to make changes like that. Once I am, I absolutely will, but until then, I'll see if I can make suggestions
